src/common/tiglcommonfunctions.cpp

```diff
@@ -1156,6 +1159,122 @@ TopoDS_Wire BuildWireFromEdges(const TopoDS_Shape& edges)
     return result;
 }
 
+opencascade::handle<Geom_BSplineCurve> ApproximateArcOfCircleToRationalBSpline(double radius, size_t nb_points, double uMin, double uMax, double y_position, double z_position)
+{
+    std::vector<gp_Pnt> arcPnts(nb_points);
+    std::vector<double> alpha = LinspaceWithBreaks(uMin, uMax, nb_points);
+    size_t index = 0;
+    for (double a : alpha){
+        arcPnts.at(index++) = (gp_Pnt(0., y_position + radius * std::cos(a), z_position + radius* std::sin(a)));
+            }
+    return tigl::CTiglPointsToBSplineInterpolation(arcPnts).Curve();
+}
+
+TopoDS_Wire BuildWireRectangle(const double heightToWidthRatio, const double cornerRadius, const double tol)
+{
+    int nb_points(0);
+
+    if(cornerRadius<0.||cornerRadius>0.5){
+        throw tigl::CTiglError("Invalid input for corner radius. Must be in range: 0 <= cornerRadius <= 0.5");
+    }
+
+    std::vector<Handle(Geom_BSplineCurve)> curves;
+
+    // build half upper line from gp_points
+    std::vector<gp_Pnt> linePntsUpperRightHalf;
+    linePntsUpperRightHalf.push_back(gp_Pnt(0.,0.,0.5*heightToWidthRatio));
+    linePntsUpperRightHalf.push_back(gp_Pnt(0.,0.5-cornerRadius,0.5*heightToWidthRatio));
+    opencascade::handle<Geom_BSplineCurve> lowerLineRightHalf = tigl::CTiglPointsToBSplineInterpolation(linePntsUpperRightHalf).Curve();
+    curves.push_back(lowerLineRightHalf);
+
+    if (!(cornerRadius == 0.0)){
+        //calculate the number of points required to maintain the minimum distance (<tol) between the bisector of two neighboring points on an arc circle
+        nb_points = std::round(M_PI/(acos(1-tol/cornerRadius)));
```

Comment:
We should never round down. We should always round up, so [`std::ceil`](https://en.cppreference.com/w/cpp/numeric/math/ceil) would be more appropriate

```suggestion
        nb_points = std::ceil(M_PI/(acos(1-tol/cornerRadius)));
```